pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java

```diff
@@ -407,11 +408,12 @@ public String cancelQuery(
         }
         return resp;
       } else if (_requestHandler.cancelQuery(Long.parseLong(id), timeoutMs, _executor, _httpConnMgr, serverResponses)) {
-          String resp = "Cancelled query: " + id;
-          if (verbose) {
-            resp += " with responses from servers: " + serverResponses;
-          }
-          return resp;
+        QueryThreadContext.setIds(Long.parseLong(id), id);
```

Comment:
Should the Ids be set at the beginning of the try block ? Both `cancelQueryByClientId` and `cancelQuery` call `queryDispatcher` etc Should the Ids be set at the beginning of the try block ? Both `cancelQueryByClientId` and `cancelQuery` call `queryDispatcher` etc

Reply:
Changed